pkg/asset/agent/image/oc.go

```diff
@@ -0,0 +1,133 @@
+package image
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/executer"
+	"github.com/sirupsen/logrus"
+	"github.com/thedevsaddam/retry"
+)
+
+const (
+	MachineOsImageName = "machine-os-images"
+	CoreOsFileName     = "/coreos/coreos-x86_64.iso"
+	OcDefaultTries     = 5
+	OcDefaltRetryDelay = time.Second * 5
+)
+
+type Config struct {
+	MaxTries   uint
+	RetryDelay time.Duration
+}
+
+type Release interface {
+	GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSecret string, platformType models.PlatformType) (string, error)
+}
+
+type release struct {
+	executer executer.Executer
+	config   Config
+}
+
+func NewRelease(executer executer.Executer, config Config) Release {
+	return &release{executer: executer, config: config}
+}
+
+const (
+	templateGetImage     = "oc adm release info --image-for=%s --insecure=%t %s"
+	templateImageExtract = "oc image extract --file=%s --path /:/%s --confirm %s"
+)
+
+// Get the CoreOS ISO from the releaseImage
+func (r *release) GetBaseIso(log logrus.FieldLogger, releaseImage string, pullSecret string, platformType models.PlatformType) (string, error) {
+
+	// Get the machine-os-images pullspec from the release and use that to get the CoreOS ISO
+	image, err := r.getImageFromRelease(log, MachineOsImageName, releaseImage, pullSecret, true)
+	if err != nil {
+		return "", err
+	}
+
+	cacheDir, err := GetCacheDir(imageDataType)
+	if err != nil {
+		return "", err
+	}
+
+	// Check if file is already cached
+	filePath, err := GetFileFromCache(CoreOsFileName, cacheDir)
+	if err != nil {
+		return "", err
+	}
+	if filePath != "" {
+		// Found cached file
+		return filePath, nil
+	}
+
+	path, err := r.extractFileFromImage(log, image, cacheDir, pullSecret, true, platformType)
+	if err != nil {
+		return "", err
+	}
+	return path, err
+}
+
+func (r *release) getImageFromRelease(log logrus.FieldLogger, imageName, releaseImage, pullSecret string, insecure bool) (string, error) {
+	cmd := fmt.Sprintf(templateGetImage, imageName, insecure, releaseImage)
+
+	log.Infof("Fetching image from OCP release (%s)", cmd)
+	image, err := execute(log, r.executer, pullSecret, cmd)
+	if err != nil {
+		return "", err
+	}
+
+	return image, nil
+}
+
+func (r *release) extractFileFromImage(log logrus.FieldLogger, image, cacheDir, pullSecret string, insecure bool, platformType models.PlatformType) (string, error) {
+
+	file := CoreOsFileName
```

Comment:
Can we add at least a TODO to use `platformType`?

Reply:
Updated to include the architecture in the coreos filename. I removed all the platformType stuff for now, we can add that back if the platform will determine which ISO to get.